# HG changeset patch
# User Rik <rik@octave.org>
# Date 1397595630 25200
#      Tue Apr 15 14:00:30 2014 -0700
# Node ID 75ec138ba53b6c92284866821f41a31a0b977169
# Parent  93d0fc8a5d868c4c34a2c68177ec8d1df06add45
Fix Z-order stacking of line markers and other graphic objects (bug #40054).

gl-render.cc (opengl_renderer::draw_line): If no zdata, use 0 for zdata rather
than 1/n where n is the number of markers to be drawn.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -1543,17 +1543,17 @@ opengl_renderer::draw_line (const line::
 
       init_marker (props.get_marker (), props.get_markersize (),
                    props.get_linewidth ());
 
       for (int i = 0; i < n; i++)
         {
           if (clip[i] == clip_ok)
             draw_marker (x(i), y(i),
-                         has_z ? z(i) : static_cast<double> (i) / n,
+                         has_z ? z(i) : 0.0,
                          lc, fc);
         }
 
       end_marker ();
     }
 
   set_clipping (props.is_clipping ());
 }
